### PR TITLE
feat: enhances S3Download to filter by traced table names

### PR DESCRIPTION
**Pull request by staheri14**
This PR updates the `S3Download` implementation to allow specifying the names of the traced tables when downloading from an S3 bucket. This enhancement enables file filtering and reduces the amount of data downloaded. It is a desirable feature when interacting with the traced data of large network tests. If no table name is provided, then `S3Download` downloads all the traced tables.

### CODE REVIEW

File: pkg/trace/fileserver.go [hunk 1]
```diff
@@ -259,7 +259,9 @@ func (lt *LocalTracer) PushAll() error {
 
 // S3Download downloads files that match some prefix from an S3 bucket to a
 // local directory dst.
-func S3Download(dst, prefix string, cfg S3Config) error {
+// tableNames is a list of traced jsonl file names to download. If it is empty, all traces are downloaded.
+// tableNames should not have .jsonl suffix.
+func S3Download(dst, prefix string, fileNames []string, cfg S3Config) error {
```
Comment (rootulp) [inline pkg/trace/fileserver.go:264]: [nit] the comment name doesn't match the argument name. 

```suggestion
// fileNames is a list of traced jsonl file names to download. If it is empty, all traces are downloaded.
// fileNames should not have .jsonl suffix.
func S3Download(dst, prefix string, fileNames []string, cfg S3Config) error {
```

**Review by rootulp — APPROVED** [review]

File: pkg/trace/fileserver.go (hunk 1 above)
Comment (staheri14) [reply pkg/trace/fileserver.go]: Right, fixed https://github.com/celestiaorg/celestia-core/pull/1374/commits/bbae134e0a8d67e5d706fcb1908dcf51461f45fd

**Review by evan-forbes — APPROVED** [review]

File: pkg/trace/fileserver.go [hunk 2]
```diff
@@ -288,37 +290,50 @@ func S3Download(dst, prefix string, cfg S3Config) error {
...
+					})
+					if err != nil {
+						f.Close()
+						continue
+					}
+					defer resp.Body.Close()
+
+					// Copy the contents of the S3 object to the local file
+					if _, err := io.Copy(f, resp.Body); err != nil {
+						return false
```
Comment (cristaloleg) [inline pkg/trace/fileserver.go:330]: Looks like `f.Close()` is missing.

**Review by rach-id — DISMISSED** [review]
utAck

**Review by rootulp — DISMISSED** [review]

File: pkg/trace/fileserver.go (hunk 2 above)
Comment (staheri14) [reply pkg/trace/fileserver.go:331]: Right, it was missing in the previous implementation as well, going to add it, nice catch! fixed it now https://github.com/celestiaorg/celestia-core/pull/1374/commits/51ad64674fa652ea20b7f298185d13a8e1855ee8

**Comment by staheri14** [comment]
The govulncheck error will be resolved by #1377

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Review by cristaloleg — APPROVED** [review]

**Pull request merged**